translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2551,6 +2558,9 @@ Value* SPIRVToLLVM::addLoadInstRecursively(
 {
     LLPC_ASSERT(pLoadPointer->getType()->isPointerTy());
 
+    // TODO: Remove this and fix lit tests to cope with fast math flags.
+    getBuilder()->clearFastMathFlags();
```

Comment:
I think we should hide it in BuilderImp instead of set it explicitly in SPIRV Reader.
If you expose more and more state, builder will be difficult to use for client.

Reply:
IRBuilder already has FMF state, which it uses when generating an FP operation. I want our Builder to work consistently with that. These TODOs are temporary; once we remove them, then the parts of the spir-v reader that use Builder will consistently get the FMF that the spir-v reader sets up to correspond to the overall mode of the shader and any decorations on the instruction. Then, once the spir-v reader uses Builder everywhere, it will all work consistently, and be easy to use for the Builder client.

Reply:
I think FMF state should only affect FP operation, so it should not affect non-FP operations regardless it is from IRBuilder or LLPC Builder.
I think it is better if LLPC builder can ignore the fast math flag for non-FP operations.
I am OK to approve this change first if it is not easy to implement it in builder.

Reply:
I believe our builder does ignore FMF for non-FP operations. Maybe I inserted this TODO and clearFastMathFlags in more places than I needed to. Is that what you're seeing? (I just inserted it everywhere that uses builder to be safe, rather than only places that use builder for an FP operation.)

Reply:
Hi Qun. Please see subsequent commits #192 and #193, which I want to land straight after this one. Can you now approve this one?

Reply:
Qun is on vacation. You can make updates if he has some suggestions. I approve this now. You can continue your work.

Reply:
Thanks Rex.